protocol/v2/ssv/validator/validator.go

```diff
@@ -86,6 +86,10 @@ func NewValidator(pctx context.Context, cancel func(), options Options) *Validat
 
 // StartDuty starts a duty for the validator
 func (v *Validator) StartDuty(logger *zap.Logger, duty *spectypes.Duty) error {
+	if ready, err := v.NodeChecker.IsReady(v.ctx); err != nil || !ready {
+		logger.Panic("nodes must be ready, otherwise there's a risk of getting slashed")
+	}
+
```

Comment:
Panic & message are correct, and [the issue i mentioned here](https://github.com/bloxapp/ssv/pull/948#issuecomment-1586248162) about `StartDuty` is no longer relevant, however I forgot to mention another issue — waiting for `StartDuty` to panic might cause us to never panic at all (for example, if the nodes crash/stuck and stop providing duties).

For that reason, we should find some other place to make this check & panic.

There are a few ways to do this:

1. After starting the prober, start another goroutine to do this check every X seconds
2. Change `ready bool` to `ready chan bool` and `IsReady()` to `IsReady() (<-chan bool)`, and send update to `ready` at every probe. With this, we don't need the `Wait` at all, we can just listen to `IsReady()` until its `true`, and then start the node and start a goroutine to listen to it until its `false` to panic
3. Accept "callback" in `NewProber` and we'll pass one that panics if ready is false

Would love to hear what you think is the safest/simplest.

Reply:
the above lines in startDuty might not be needed because we added the periodic check. 
@moshe-blox do you think there is anything else that should be done?

Reply:
@nkryuchkov ,
I feel like we can keep the wait. but the bottom line is that we can't panic only at `StartDuty` because probably if we got a duty, we are synced, and if we are not synced we're not getting any duties.
So its not the optimal place to do this check, it might be good thing IMO to go with moshe's option 3, pass a callback into the prober constructor that will determine what do to when the nodes are not ready. we need somehow to make it active only after the first wait though.

Reply:
@y0sher I see, thanks. Implemented. Could you please review again?